### ISSUE TITLE
`IdentResult.FINISHED` is triggered twice in `IDNowSDKResultListener` 

### ISSUE THREAD

**Issue by juliancadi** [idnow/de.idnow.ios.sdk.spm#19]
**Version:** `5.0.1`
**Installation:** Manual
**Type:** Without NFC
**Device:** iPhone 11 Pro Max
**OS Version:** `iOS 16.1`

---
The callback `IDNowSDKResultListener` is triggering the result `IdentResult.FINISHED` **_twice_** in the following implementation:

```swift
IDNowSDK.shared.start(
    token: "MYTOKEN",
    fromViewController: myViewController
) { result, statusCode, message in
    switch result {                
    case .FINISHED:
        print("[IDNow] FINISHED with result(\(result.rawValue)) - statusCode(\(statusCode.rawValue)) - message(\(message))")

    case .CANCELLED:
        print("[IDNow] CANCELLED with result(\(result.rawValue)) - statusCode(\(statusCode.rawValue)) - message(\(message))")
        
    case .ERROR:
        print("[IDNow] ERROR with result(\(result.rawValue)) - statusCode(\(statusCode.rawValue)) - message(\(message))")
        
    @unknown default:
        print("[IDNow] DEFAULT with result(\(result.rawValue)) - statusCode(\(statusCode.rawValue)) - message(\(message))")
    }
}
```

Output ([Full stack trace](https://github.com/idnow/de.idnow.ios.sdk.spm/files/10086217/idnow.log)):
```log
[IDNow] FINISHED with result(0) - statusCode(10) - message() // 1
[IDNow] FINISHED with result(0) - statusCode(10) - message(Success) // 2
```

1. First time right after `IDNowSDK.shared.start` call and IDNowSDK view is loaded (Not expected, not happening in previous versions ❌)
2. Second time when IDNow AutoIdent flow is sucessfully completed (Expected ✅)

As a workaround we are validating `message == "Success"` to make sure the result really corresponds to the second scenario above where the flow successfully completes:

```swift
   ...
    case .FINISHED:
        guard message == "Success" else { 
            print("[IDNow] FINISHED - WRONG STATE
            return
        }
        print("[IDNow] FINISHED with result(\(result.rawValue)) - statusCode(\(statusCode.rawValue)) - message(\(message))")
   ...
```

However I don't think this double `FINISHED` callback is intentional and it wasn't happening in versions 5.0.0 or previous.

**Comment by deryck-tomiuc-idnow** [comment]
Hello @juliancadi,
Thank you for sharing this with us. We will be investigating and have a fix for it as soon as possible.
Thank you!

**Comment by juliancadi** [comment]
@deryck-tomiuc-idnow any update on this maybe?

**Comment by nrj** [comment]
Also experiencing this issue.. the workaround above works.

**Comment by deryck-tomiuc-idnow** [comment]
Hello @juliancadi, @nrj. thank you for your patience on this topic. We have released a new version (5.0.4) and it contains a fix for this issue. Please let me know if you are having any issues with it. Thank you!

**Comment by juliancadi** [comment]
Version 5.0.4 solves this
